### ISSUE TITLE
ControlNet with SAM mask condition

### ISSUE THREAD

**Issue by gasvn** [lllyasviel/ControlNet#333]
Thanks for the great ControlNet. We have trained a [ControlNet based on the SAM segmentation mask](https://github.com/sail-sg/EditAnything) for fine-grained image generation.
This model can help to use SAM for image generation and editing. We hope this side project can help with more interesting applications. Looking for contributions and suggestions!
https://github.com/sail-sg/EditAnything

**Comment by lllyasviel** [comment]
how is the label color computed? random color?

**Comment by gasvn** [comment]
Yes, the label color is randomly generated. We didn't use the random color for training, it's just for visualization.

**Comment by lllyasviel** [comment]
cool! is it fine tuned from canny or scribble model?

**Comment by sethupavan12** [comment]
@alelordelo I was actually going to ask if this is possible as I plan to train the control net on my dataset comprised of segmentations and prompts. 

I assume it is currently not possible to train with segmentations then?

**Comment by alelordelo** [comment]
I think not now... : /

**Comment by alelordelo** [comment]
@gasvn @lllyasviel @sethupavan12 , did you do any progress on the segmented training?

**Comment by sethupavan12** [comment]
@alelordelo Yes....in a manner of speaking. Basically I simplified my problem.
Initially, I had a segmentation image like the following


![image](https://github.com/lllyasviel/ControlNet/assets/60856766/26c3699e-d50e-4169-a76a-d947341bc03b)

But then I realised I didn't need the other body parts as my main focus is just clothing/ body area as I was working on virtual try on problem.

So I masked out the useless areas and masked the areas I want to train on.

Masked area cut from main image looks like
![image](https://github.com/lllyasviel/ControlNet/assets/60856766/97a671e4-c139-4db4-b546-5892d8ad248a)
![image](https://github.com/lllyasviel/ControlNet/assets/60856766/5fd33c45-d1fd-472e-b5e7-1ffb5d0f4322)

Then trained with the prompts the original image has something like _Woman wearing blue skirt_ etc.

![image](https://github.com/lllyasviel/ControlNet/assets/60856766/8f3160fc-34c8-43c1-a754-08a680af3a6c)


Might not apply to everyone but yeah thats what I ended up doing.

**Comment by sethupavan12** [comment]
But ofc removing the "useless" area would mean I can't generate as low-level I want, or specific areas. Segmentation training would save you a lot of time. but if u got time, maybe u can just mask each part and train each part ? I have not tried this so idk

**Comment by alelordelo** [comment]
Ok, so basically you trained:
prompt + segmentation map (all masks) = image

And then inference:
prompt + masked area = output image

Is that it?

I guess this will work for dresses as your example, but not for full looks (shirt + pants + shoes)?
For this case, you would need to match prompt with mask region I think?
